plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/timeline/HudiActiveTimeline.java

```diff
@@ -85,7 +85,8 @@ public Optional<byte[]> readCompactionPlanAsBytes(HudiInstant instant)
 
     private Location getInstantFileNamePath(String fileName)
     {
-        return Location.of(fileName.contains(SCHEMA_COMMIT_ACTION) ? metaClient.getSchemaFolderName() : metaClient.getMetaPath().path()).appendPath(fileName);
+        Location schemaFolder = metaClient.getMetaPath().appendPath(HudiTableMetaClient.SCHEMA_FOLDER_NAME);
+        return (fileName.contains(SCHEMA_COMMIT_ACTION) ? schemaFolder : metaClient.getMetaPath()).appendPath(fileName);
```

Comment:
There's no build `schemaFolder` object when `fileName` doesn't contain `SCHEMA_COMMIT_ACTION`. This can be something like:
```java
    private Location getInstantFileNamePath(String fileName)
    {
        Location metaPath = metaClient.getMetaPath();
        if (fileName.contains(SCHEMA_COMMIT_ACTION)) {
            return metaPath.appendPath(HudiTableMetaClient.SCHEMA_FOLDER_NAME);
        }
        return metaPath.appendPath(fileName);
    }
```

Reply:
Thanks for the observation, I have changed the logic accordingly.